packages/application-shell/src/hooks/use-applications-menu/use-applications-menu.ts

```diff
@@ -27,7 +27,9 @@ export type MenuLoaderResult<Key extends MenuKey> = Key extends 'appBar'
   : never;
 export type Config<Key extends MenuKey> = {
   environment: TApplicationContext<{}>['environment'];
-  queryOptions?: QueryFunctionOptions;
```

Comment:
Somehow TS was getting a bit confused, so I decided to be explicit in which query options are being used from our side.